docs/app-check/debug-provider.md

```diff
@@ -16,9 +18,165 @@ Warning: The debug provider allows access to your Firebase resources from
+          AppCheck.setAppCheckProviderFactory(providerFactory)
+      #endif
+
+          GeneratedPluginRegistrant.register(with: self)
+          return super.application(application, didFinishLaunchingWithOptions: launchOptions)
+        }
+      }
+      ```
+
+      Tip: Confirm in your Xcode project's Build Settings that it passes the
+      `DEBUG` flag to the Swift compiler.
+
+    * {Objective-C}
+
+      ```objc
+      #import "AppDelegate.h"
+      #import "GeneratedPluginRegistrant.h"
+      @import Firebase;  // Add the Firebase import.
+
+      @implementation AppDelegate
+
+      - (BOOL)application:(UIApplication *)application
+          didFinishLaunchingWithOptions:(NSDictionary *)launchOptions {
+        // Use the debug provider in Debug builds:
+      #if DEBUG
+        FIRAppCheckDebugProviderFactory *providerFactory = [[FIRAppCheckDebugProviderFactory alloc] init];
+        [FIRAppCheck setAppCheckProviderFactory:providerFactory];
+      #endif
+        [GeneratedPluginRegistrant registerWithRegistry:self];
+        // Override point for customization after application launch.
+        return [super application:application didFinishLaunchingWithOptions:launchOptions];
+      }
+
+      @end
+      ```
+
+1.  Enable debug logging in your Xcode project (v11.0 or newer):
+
+    1.  Open **Product > Scheme > Edit scheme**.
+    1.  Select **Run** from the left menu, then select the **Arguments** tab.
+    1.  In the **Arguments Passed on Launch** section, add `-FIRDebugEnabled`.
+
+1.  Open `ios/Runner.xcworkspace` with Xcode and run your app in the Simulator.
+    A local debug token will be logged to the debug output when Firebase tries
+    to send a request to the backend. For example:
+
+    <pre>[Firebase/AppCheck][I-FAA001001] Firebase App Check Debug Token:
+    123a4567-b89c-12d3-e456-789012345678</pre>
+
+{# Google-internal common file: #}
+<<../_includes/manage-debug-tokens.md>>
+
+After you register the token, Firebase backend services will accept it as valid.
+
+Because this token allows access to your Firebase resources without a
+valid device, it is crucial that you keep it private. Don't commit it to a
+public repository, and if a registered token is ever compromised, revoke it
+immediately in the Firebase console.
+
+## Android
+
+To use the debug provider while running your Flutter app in an android environment,
+implement the following:
+
+```dart
+import 'package:flutter/material.dart';
+import 'package:firebase_core/firebase_core.dart';
+
+// Import the firebase_app_check plugin
+import 'package:firebase_app_check/firebase_app_check.dart';
+
+Future<void> main() async {
+  WidgetsFlutterBinding.ensureInitialized();
+  await Firebase.initializeApp();
+  await FirebaseAppCheck.instance.activate(
+    webRecaptchaSiteKey: 'recaptcha-v3-site-key',
+    // Set the androidDebugProvider to `true`
+    androidDebugProvider: true,
+  );
+  runApp(App());
+}
+
```

Comment:
Hey @kevinthecheung, could you let me know if this documentation update for android is acceptable? We'll get it merged in after we've merged this [android debugProvider](https://github.com/firebase/flutterfire/pull/9412) feature. Thanks!

Reply:
This commit for ease of reference: https://github.com/firebase/flutterfire/pull/9445/commits/55bb22327ea614705d918132eadbb2210accdbcf

Reply:
Looks great! Thanks.